packages/common/test/objects/deepFilter.test.ts

```diff
@@ -0,0 +1,113 @@
+import { deepFilter } from "../../src/objects/deepFilter";
+
+describe("deepFilter:", () => {
+  describe("predicate:", () => {
+    it("filters entries in an array", () => {
+      const test = ["a", "b", "c"];
+      const predicate = (obj: any) => obj !== "b";
+      const chk = deepFilter(test, predicate);
+
+      expect(chk.length).toBe(2);
+      expect(chk).toEqual(["a", "c"]);
+    });
+    it("filters object entries in array", () => {
+      const test = [{ notid: "a" }, { id: "b" }, { id: "c" }];
+      const predicate = (obj: any) => obj?.id !== "b";
+      const chk = deepFilter(test, predicate);
+
+      expect(chk.length).toBe(2);
+      expect(chk).toEqual([{ notid: "a" }, { id: "c" }]);
+    });
+    it("filters simple objects", () => {
+      const test = { id: "b" };
```

Comment:
This one is confusing - can you add example where it would filter something out?